scripts/generate/test_generate.py

```diff
@@ -95,12 +95,20 @@ def test_generate(
             "The input prompt is empty, model will respond from a empty sequence."
         )
 
-    utils.set_determinism(seed)
+    if seed is not None:
+        torch.manual_seed(seed)
+        # PYTHONHASHSEED can be a decimal number in the range [0, 2**32 - 1]
+        os.environ["PYTHONHASHSEED"] = str(seed % 2**32)
+        torch.use_deterministic_algorithms(True)
+        torch.backends.cudnn.deterministic = True
+        torch.backends.cudnn.benchmark = False
+        # env var for deterministic CuBLAS
+        # https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html
+        os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"
```

Comment:
Why is the `set_determinism()` removed from the test?

Reply:
This call of `set_determinism()` was added before @wconstab's PR to make RNG right, which broke this line. Given how `seed` is explicitly used here, I'm not sure if the code over there would indicate (would it still be correct?). Need @XilunWu's help on understanding more.

This PR tries to restore the behavior before the "BC-breaking change" and guard on its running.

Reply:
why is the set_determinism change for correct RNG affecting this code?  I would hope that calling `set_determinism(seed, deterministic=True) would be equivalent to some manual stuff done here.  What is the issue?

Reply:
discussed offline; refactored to use `set_determinism`